src/java.base/share/classes/java/time/Instant.java

```diff
@@ -1164,6 +1164,12 @@ private long nanosUntil(Instant end) {
         return Math.addExact(totalNanos, end.nanos - nanos);
     }
 
+    private long microsUntil(Instant end) {
+        long secsDiff = Math.subtractExact(end.seconds, seconds);
+        long totalMicros = Math.multiplyExact(secsDiff, NANOS_PER_SECOND / 1000);
+        return Math.addExact(totalMicros, (end.nanos - nanos) / 1000);
```

Comment:
Can you define NANOS_PER_MICRO, the others conversions use predefined constants.

Reply:
Defined it as a private field in `Instant`.